metricbeat/module/system/core/core_test.go

```diff
@@ -41,10 +42,13 @@ func TestFetch(t *testing.T) {
 }
 
 func TestData(t *testing.T) {
+	if !*flags.DataFlag {
+		return
+	}
 	f := mbtest.NewReportingMetricSetV2Error(t, getConfig())
 
 	mbtest.ReportingFetchV2Error(f)
-	time.Sleep(500 * time.Millisecond)
+	time.Sleep(5 * time.Second)
```

Comment:
That's one way to slow down CI. Why do we need this big delay in an unit test?

Reply:
That was mostly so I could generate a reasonably accurate `data.json` file. However, I assumed the `TestData` tests weren't run in CI?

Reply:
Correct, given the `if !*flags.DataFlag` above this code won't be run in CI.

Reply:
> Correct, given the if !*flags.DataFlag above this code won't be run in CI.

Can you confirm for sure that these are not run?

What are these tests good for if we don't run them?

Reply:
> Can you confirm for sure that these are not run?

The data flag defaults to false, and afaik is not set in CI: https://github.com/elastic/beats/blob/7426f67d658dcddc4a28fae9e5aec0c6c37be895/metricbeat/mb/testing/flags/flags.go#L25

> What are these tests good for if we don't run them?

More than tests these `TestData` tests are scripts used to generate the example data files, and they are called manually when we want to update these files. In most cases they do basically the same as `TestFetch`, but with less checks, and sometimes ensuring that the data files written are somehow meaningful.

The method to write the data files is the one that uses to leverage this `DataFlag` to skip the test if not set, as in here: https://github.com/elastic/beats/blob/7426f67d658dcddc4a28fae9e5aec0c6c37be895/metricbeat/mb/testing/data_generator.go#L55-L58

I agree that this is not ideal, because some code is always executed in these `TestData` tests before skipping, but this is out of the scope of this PR, and it will be different with integrations in any case.